lnwire/update_add_htlc.go

```diff
@@ -132,3 +158,58 @@ func (c *UpdateAddHTLC) MsgType() MessageType {
 func (c *UpdateAddHTLC) TargetChanID() ChannelID {
 	return c.ChanID
 }
+
+// EndorsedHTLCExperimental is the TLV number for experimental HTLC
+// endorsement signaling.
+// See: https://github.com/lightning/blips/pull/27
+const EndorsedHTLCExperimental tlv.Type = 65555
+
+// EndorsementSignal is an alias that allows implementing the RecordProducer
+// interface on a boolean.
+type EndorsementSignal bool
+
+// NewEndorsedRecord creates a TLV record for a single-byte endorsement record.
+func (e *EndorsementSignal) Record() tlv.Record {
+	// Note: byte is an alias for uint8 in go, so we can re-use its
+	// encoding/decoding functions then convert back to our boolean value.
+	return tlv.MakeStaticRecord(
+		EndorsedHTLCExperimental, e, 1,
+		func(w io.Writer, val interface{}, buf *[8]byte) error {
+			if v, ok := val.(*EndorsementSignal); ok {
+				// Convert boolean value to a single byte.
+				var endorsedByte byte
+				if *v {
+					endorsedByte = 1
+				}
+
+				return tlv.EUint8(w, &endorsedByte, buf)
+			}
+
+			return tlv.NewTypeForEncodingErr(val, "*bool")
+		},
+		func(r io.Reader, val interface{}, buf *[8]byte,
+			l uint64) error {
+
+			if v, ok := val.(*EndorsementSignal); ok && l == 1 {
+				var endorsedByte byte
+
+				err := tlv.DUint8(r, &endorsedByte, buf, l)
+				if err != nil {
+					return err
+				}
+
+				// Interpret any non-zero value as a positive
+				// endorsement.
+				if endorsedByte == 0x0 {
```

Comment:
I think decoding any non-zero byte as positive endorsement can be written in blip draft.

If you wish to have `endorsedByte` receivers whatever the implementation interpreting results in the same sense.